Add Color class and useful color globals

Halithor opened the pull request:
Implement a Color class which wraps a simple RGBA color. Color can convert to text code and convert to player color names. Add useful color related globals for looking up a player's color and color name.

Is this a proper directory location for this? The `globals` directory didn't seem appropriate because of how it's exported.

Is this how global arrays should be exported? Usage is similar to the `Players` array.

Yes, the way I typed the color's components is odd, but it allows Typescript's type system to enforce the assigned values.

cipherxof commented:
Some suggestions:

1. `helper` -> `utils`
2. `get playercolor()` -> `get playerColor()`
3. You should be commenting methods using the TSDoc standard so your descriptions can integrate nicely with VSCode and other IDEs.

Also, here are the exact RGB codes taken directly from the game files (found in gluemanager).

```ts
t.PlayerColor = [
    { r: 255, g: 3, b: 3 },
    { r: 0, g: 66, b: 255 },
    { r: 28, g: 230, b: 185 },
    { r: 84, g: 0, b: 129 },
    { r: 255, g: 252, b: 0 },
    { r: 254, g: 138, b: 14 },
    { r: 32, g: 192, b: 0 },
    { r: 229, g: 91, b: 176 },
    { r: 149, g: 150, b: 151 },
    { r: 126, g: 191, b: 241 },
    { r: 16, g: 98, b: 70 },
    { r: 78, g: 42, b: 4 },
    { r: 155, g: 0, b: 0 },
    { r: 0, g: 0, b: 195 },
    { r: 0, g: 234, b: 255 },
    { r: 190, g: 0, b: 254 },
    { r: 235, g: 205, b: 135 },
    { r: 248, g: 164, b: 139 },
    { r: 191, g: 255, b: 128 },
    { r: 220, g: 185, b: 235 },
    { r: 80, g: 79, b: 85 },
    { r: 235, g: 240, b: 255 },
    { r: 0, g: 120, b: 30 },
    { r: 164, g: 111, b: 51 },
    { r: 45, g: 45, b: 45 },
    { r: 45, g: 45, b: 45 },
    { r: 45, g: 45, b: 45 },
    { r: 45, g: 45, b: 45 }
]
```

Probably can't notice a difference visually between the values you used and those, but might as well be exact.

Halithor commented:
1. Done
2. I kept the capitalization the same as the type that it's returning (the native `playercolor`).  Either way, updated.
3. Done. I write too much Go.

So many off by ones in the colors. The player color black was really wrong though. Updated all the player colors values to match above.
